pet-store-react-hooks/src/Home/Home.test.tsx

```diff
@@ -0,0 +1,211 @@
+  expect(cardHeader).toBeInTheDocument();
+});
+
+test('Add pet button is disabled while fetching pets and then enabled', async () => {
+  const waitHandle = new WaitHandle();
+
+  server.resetHandlers(
+    http.get(`${apiBaseUrl}/pet/kinds`, async () => {
+      await waitHandle.wait();
+      return HttpResponse.json(petKinds);
+    }),
+    http.get(`${apiBaseUrl}/pet/all`, () => HttpResponse.json(pets))
+  );
+
+  render(<Home />);
+
+  const addPetButton = await screen.findByRole('button', { name: 'Add pet' });
+  expect(addPetButton).toBeDisabled();
+
+  const loadingIndicator = await screen.findByLabelText('loading-indicator');
+  waitHandle.release();
+  await waitForElementToBeRemoved(loadingIndicator);
+
+  expect(addPetButton).toBeEnabled();
+});
+
+test('The table is displayed, columns count and text inside of them is correct', async () => {
+  render(<Home />);
+
+  const table = await screen.findByRole('table');
+  expect(table).toBeInTheDocument();
+
+  const headerCells = await screen.findAllByRole('columnheader', {
+    hidden: true,
+  });
+  expect(headerCells).toHaveLength(5);
+
+  expect(headerCells[0]).toHaveTextContent('#');
+  expect(headerCells[1]).toHaveTextContent('Name');
+  expect(headerCells[2]).toHaveTextContent('Added date');
+  expect(headerCells[3]).toHaveTextContent('Kind');
+  expect(headerCells[4]).toHaveTextContent('');
+});
+
+test('All table rows are rendered, cell values are visualized correctly and each row has the View/Edit and Delete button', async () => {
+  const waitHandle = new WaitHandle();
+
+  server.resetHandlers(
+    http.get(`${apiBaseUrl}/pet/kinds`, () => HttpResponse.json(petKinds)),
+    http.get(`${apiBaseUrl}/pet/all`, async () => {
+      await waitHandle.wait();
+      return HttpResponse.json(petsList);
+    })
+  );
+
+  render(<Home />);
+
+  const loadingIndicator = await screen.findByLabelText('loading-indicator');
+  waitHandle.release();
+  await waitForElementToBeRemoved(loadingIndicator);
+
+  const tableRows = screen.getAllByRole('row').slice(1);
+  expect(tableRows).toHaveLength(petsList.length);
+
+  const firstRowCells = within(tableRows[0]).getAllByRole('cell');
+
+  expect(firstRowCells[0]).toHaveTextContent('44');
+  expect(firstRowCells[1]).toHaveTextContent('Kenny');
+  expect(firstRowCells[2]).toHaveTextContent('27 Oct 2022');
+  expect(firstRowCells[3]).toHaveTextContent('Parrot');
+
+  const firstRowButtons = within(firstRowCells[4]).getAllByRole('button');
+  expect(firstRowButtons).toHaveLength(2);
+  expect(firstRowButtons[0]).toHaveTextContent('View / Edit');
+  expect(firstRowButtons[1]).toHaveTextContent('Delete');
+});
+
+test('New pet modal is shown on Add pet button click and then closed', async () => {
+  const user = userEvent.setup();
+  render(<Home />);
+
+  const addPetButton = await screen.findByRole('button', { name: 'Add pet' });
+  await user.click(addPetButton);
+
+  const modal = await screen.findByRole('dialog');
```

Comment:
find it by name as well, findByRole('dialog', { name: 'Add pet modal' })

And for the rest as well.